published/external/xdpapi.h

```diff
@@ -474,6 +500,53 @@ XdpOpenApi(
     return _Status;
 }
 
+inline
+_IRQL_requires_(PASSIVE_LEVEL)
+NTSTATUS
+XdpOpenApi(
+    _In_ UINT32 _XdpApiVersion,
+    _In_opt_ VOID *_ClientContext,
+    _In_opt_ XDP_API_ATTACH_FN *_ClientAttach,
+    _In_opt_ XDP_API_DETACH_FN *_ClientDetach,
+    _In_ const XDP_API_CLIENT_DISPATCH *_XdpApiClientDispatch,
+    _In_opt_ const INT64 *_TimeoutMs,
+    _Out_ XDP_API_CLIENT *_ApiContext,
+    _Out_ const XDP_API_PROVIDER_DISPATCH **_XdpApiProviderDispatch,
+    _Out_ const XDP_API_PROVIDER_BINDING_CONTEXT **_XdpApiProviderContext
+    )
+{
+    NTSTATUS Status;
+    LARGE_INTEGER Timeout100Ns;
+    LARGE_INTEGER *Timeout100NsPtr = NULL;
+    if (_TimeoutMs && *_TimeoutMs > 0) {
```

Comment:
I think we should support both Timeout == NULL and Timeout == 0, because these have different meanings. NULL means wait forever, and 0 means never wait.

Reply:
NULL is supported by this code, Timeout100NsPtr stay NULL then used for `KeWaitForSingleObject`
So this will be `if (_TimeoutMs && *_TimeoutMs >= 0)`